### PR TITLE
Review created.

### PR DESCRIPTION
**Pull request by aldos2002**

### CODE REVIEW

File: src/main/java/com/epam/dao/UserDao.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by Almas_Doskozhin
+ * on 4/2/2016.
+ */
+public class UserDao {
```
Comment (sshogunn) [inline src/main/java/com/epam/dao/UserDao.java:24]: Looks like your Persistence layer has too many responsibilities.
Keep in mind, that it must be responsible only for saving data in database

File: src/main/java/com/epam/rest/HelloWorldService.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/users")
+public class HelloWorldService {
+    private UserDao userDao = new UserDao();
+
+    @POST
+    @Path("/create")
```
Comment (sshogunn) [inline src/main/java/com/epam/rest/HelloWorldService.java]: Please read REST Services Architecture
Your REST services paths look like just Servlets paths.

Comment (sshogunn) [inline src/main/java/com/epam/rest/HelloWorldService.java]: You don't need to mention that it's _create_ or _delete_ you have REST methods for that - @POST, @DELETE, @PUT, @GET

File: src/main/java/com/epam/rest/HelloWorldService.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @GET
+    @Path("/read/{userid}")
+    @Produces(MediaType.APPLICATION_XML)
+    public User readUser(@PathParam("userid") String user) {
+        return userDao.readUser(user);
+    }
+
+    @POST
```
Comment (sshogunn) [inline src/main/java/com/epam/rest/HelloWorldService.java]: @PUT and @POST can help you  for _create_ and _update_ services

File: src/main/java/com/epam/rest/HelloWorldService.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Path("/update/{userid}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public User updateUser(@PathParam("userid") String userid, User user) {
+        int result = userDao.updateUser(userid, user);
+        if (result == 1) {
```
Comment (sshogunn) [inline src/main/java/com/epam/rest/HelloWorldService.java]: I cannot get it)
Why you check it, if you return the user object anyway?

File: src/main/java/com/epam/rest/HelloWorldService.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            return user;
+        }
+        return user;
+    }
+
+    @DELETE
+    @Path("/delete/{userid}")
```
Comment (sshogunn) [inline src/main/java/com/epam/rest/HelloWorldService.java]: DRY - delete in @DELETE and in the path

File: src/main/java/com/epam/rest/HelloWorldService.java (hunk 2 above)
Comment (aldos2002) [inline src/main/java/com/epam/rest/HelloWorldService.java]: Got rid of extra info in path.

File: src/main/java/com/epam/rest/HelloWorldService.java (hunk 3 above)
Comment (aldos2002) [inline src/main/java/com/epam/rest/HelloWorldService.java]: Used @PUT for update.

File: src/main/java/com/epam/rest/HelloWorldService.java (hunk 4 above)
Comment (aldos2002) [inline src/main/java/com/epam/rest/HelloWorldService.java]: Removed check.

File: src/main/java/com/epam/rest/HelloWorldService.java (hunk 5 above)
Comment (aldos2002) [inline src/main/java/com/epam/rest/HelloWorldService.java]: Got rid of extra info in Path

**Pull request merged**